### PR TITLE
Release v0.5.1: canonical maintenance round + AssemblyVersion fix

### LINKED ISSUE #104: [Maintenance] cleanup: Resolve template drift
## Scope

This repo has drifted from the canonical repo-template. Run the template-drift-scan, review each drifted or missing file, and re-sync the files that should track the template (workflows, configs, scripts, analyzer settings).

## Acceptance criteria

template-drift-scan reports no unexpected drift for this repo; any intentional deviation is documented.

## Notes

- Tier-1 foundation task — part of fleet initiative **C1: Resolve template drift**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #107: [Maintenance] cleanup: Version metadata consistency
## Scope

Audit Version, AssemblyVersion, FileVersion and InformationalVersion against the latest git tag. Fix hardcoded or stale values (for example AssemblyVersion stuck at 1.0.0).

## Acceptance criteria

Version metadata is consistent and matches the release tag scheme.

## Notes

- Tier-1 foundation task — part of fleet initiative **C4: Version metadata consistency**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #108: [Maintenance] cleanup: Enable nullable reference types
## Scope

Enable Nullable=enable for this repo and resolve the resulting nullable warnings.

## Acceptance criteria

Nullable reference types enabled; build clean with no nullable warnings.

## Notes

- Tier-1 foundation task — part of fleet initiative **C5: Enable nullable reference types**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #110: [Maintenance] cleanup: Run an AI full code-review pass
## Scope

Run a full AI-assisted code review of this repo. File each finding as its own Tier-2 maintenance-task sub-issue in the appropriate category.

## Acceptance criteria

Code review complete; findings filed as categorized sub-issues; this issue links them.

## Notes

- Tier-1 foundation task — part of fleet initiative **C7: Run an AI full code-review pass**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #112: [Maintenance] performance: Stand up a BenchmarkDotNet baseline project
## Scope

Add a BenchmarkDotNet project covering this repo primary code paths and capture a baseline run.

## Acceptance criteria

A benchmarks/ BDN project exists, builds, runs, and a baseline result is recorded.

## Notes

- Tier-1 foundation task — part of fleet initiative **P1: Stand up a BenchmarkDotNet baseline project**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #113: [Maintenance] performance: Publish benchmark-result graphs to gh-pages
## Scope

Add the benchmarks.yaml workflow that runs BenchmarkDotNet and publishes result charts to gh-pages (see the add-benchmarks-workflow skill and the ETL-FixedWidth reference).

## Acceptance criteria

Benchmark charts publish to the repo dev/bench gh-pages path on push to main.

## Notes

- Tier-1 foundation task — part of fleet initiative **P2: Publish benchmark-result graphs to gh-pages**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #114: [Maintenance] testing: Publish code-coverage graphs to docs pages
## Scope

Publish a code-coverage trend graph to this repo documentation pages.

## Acceptance criteria

A coverage graph is visible on the repo gh-pages docs site and updates over time.

## Notes

- Tier-1 foundation task — part of fleet initiative **T1: Publish code-coverage graphs to docs pages**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #116: [Maintenance] testing: Add Stryker mutation testing
## Scope

Add Stryker.NET mutation testing to validate that the test suite actually asserts behavior.

## Acceptance criteria

Stryker runs for this repo; mutation score recorded; surviving mutants triaged.

## Notes

- Tier-1 foundation task — part of fleet initiative **T3: Add Stryker mutation testing**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #117: [Maintenance] docs: Standardize README structure
## Scope

Bring this repo README to the standard structure — consistent sections plus build, coverage, and NuGet status badges.

## Acceptance criteria

README follows the standard section layout and carries the standard badges.

## Notes

- Tier-1 foundation task — part of fleet initiative **D1: Standardize README structure**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #118: [Maintenance] docs: Create a CHANGELOG if missing
## Scope

Add a CHANGELOG in Keep a Changelog style if this repo lacks one; seed it from release history.

## Acceptance criteria

CHANGELOG.md exists and reflects released versions.

## Notes

- Tier-1 foundation task — part of fleet initiative **D2: Create a CHANGELOG if missing**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #120: [Maintenance] docs: Audit README for accuracy
## Scope

Verify the README content is current and correct — install steps, API examples, links, and version references.

## Acceptance criteria

README content verified accurate against the current code and package.

## Notes

- Tier-1 foundation task — part of fleet initiative **D4: Audit README for accuracy**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #121: [Maintenance] docs: Audit other Markdown files for accuracy
## Scope

Verify CONTRIBUTING.md, SECURITY.md, CODE_OF_CONDUCT.md and docs/*.md are current and correct.

## Acceptance criteria

All non-README Markdown verified accurate; stale content fixed.

## Notes

- Tier-1 foundation task — part of fleet initiative **D5: Audit other Markdown files for accuracy**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #122: [Maintenance] docs: Verify documentation builds correctly on release
## Scope

Verify the docfx documentation builds and deploys correctly as part of the release workflow.

## Acceptance criteria

A release produces correctly built docs on gh-pages with no docfx build errors.

## Notes

- Tier-1 foundation task — part of fleet initiative **D6: Verify documentation builds correctly on release**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #123: [Maintenance] docs: Verify previous-version documentation is preserved
## Scope

Confirm documentation for prior released versions is preserved on gh-pages and not overwritten by new releases.

## Acceptance criteria

Versioned docs for all prior releases remain accessible.

## Notes

- Tier-1 foundation task — part of fleet initiative **D7: Verify previous-version documentation is preserved**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #124: [Maintenance] docs: Improve the documentation version selector
## Scope

Implement a documentation **version selector dropdown at the top of every docs page**, matching the working implementation on DateTime-Extensions.

See it live: https://chris-wolfgang.github.io/DateTime-Extensions/ — the dropdown sits in the page header and lets readers jump between every released version of the docs.

## What "the DateTime-Extensions way" means concretely

The picker is built from four canonical assets that should already exist in this repo via the C1 / D8 canonical sync:

| Asset | Purpose |
|---|---|
| `docfx_project/public/version-picker.js` | Reads `versions.json`, renders the dropdown, handles selection |
| `docfx_project/versions.json` | The list of versions to populate the dropdown |
| `.github/version-picker-template.html` | HTML shell for the picker UI |
| `docfx_project/docfx.json` (`globalMetadata._appFooter`) | Injects the picker script into every rendered page |

The reference implementation in DateTime-Extensions wires all four together. Confirm each is present in this repo and matches the canonical pattern.

## Acceptance criteria

1. Dropdown is **visible at the top of every docs page** on the gh-pages site for this repo (not hidden behind a menu, not absent).
2. Dropdown lists **every published version tag** for this repo, with the latest preselected.
3. Selecting a version navigates to the corresponding `/versions/v<n>/` path.
4. The unversioned landing page (`/`) redirects to `/versions/<latest>/`.
5. `versions.json` is updated on each release by `release.yaml` / `docfx.yaml` (this is part of the D6 preservation guard + D8 picker handshake — verify the workflow path actually publishes the new version into the JSON).

## Notes

- Tier-1 foundation task — part of fleet initiative **D8: Improve the documentation version selector**.
- The canonical assets were rolled out via the C1/D8 fleet sync; what remains is repo-specific: verify the picker actually renders on this repo's gh-pages, that `versions.json` lists the right tags, and that the per-release deploy populates new entries correctly.
- If the picker is invisible on this repo's docs site despite the assets being present, the most common causes are (a) `versions.json` is empty / stale, (b) the `_appFooter` injection got stripped during a per-repo `docfx.json` edit, or (c) older versions weren't deployed to `/versions/v<n>/` paths on `gh-pages`.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #125: [Maintenance] security: Enable CodeQL security-extended query pack
## Scope

Switch this repo CodeQL workflow to the security-extended query pack for deeper analysis.

## Acceptance criteria

codeql workflow uses the security-extended queries; scan passes.

## Notes

- Tier-1 foundation task — part of fleet initiative **S1: Enable CodeQL security-extended query pack**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #126: [Maintenance] security: GitHub Actions hardening
## Scope

SHA-pin all third-party GitHub Actions, add least-privilege permissions blocks to every workflow, and consider adding the OpenSSF Scorecard action.

## Acceptance criteria

All third-party actions SHA-pinned; every workflow declares minimal permissions.

## Notes

- Tier-1 foundation task — part of fleet initiative **S2: GitHub Actions hardening**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #128: [Maintenance] API: Add PublicApiAnalyzers for breaking-change detection
## Scope

Add Microsoft.CodeAnalysis.PublicApiAnalyzers with PublicAPI.Shipped.txt and PublicAPI.Unshipped.txt to catch unintended public-API changes.

## Acceptance criteria

PublicApiAnalyzers active; public API surface tracked in PublicAPI.*.txt.

## Notes

- Tier-1 foundation task — part of fleet initiative **A1: Add PublicApiAnalyzers for breaking-change detection**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #129: [Maintenance] CI/CD: NuGet package quality
## Scope

Improve the published package: enable SourceLink, deterministic and CI builds, symbol packages (.snupkg), and complete package metadata (description, tags, icon, PackageReadmeFile, SPDX license expression).

## Acceptance criteria

Package has SourceLink, deterministic build, .snupkg symbols, and complete metadata.

## Notes

- Tier-1 foundation task — part of fleet initiative **CI1: NuGet package quality**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #130: [Maintenance] CI/CD: Add the github-actions ecosystem to Dependabot
## Scope

Add a github-actions package-ecosystem entry to .github/dependabot.yml so workflow action versions stay current.

## Acceptance criteria

dependabot.yml watches the github-actions ecosystem.

## Notes

- Tier-1 foundation task — part of fleet initiative **CI2: Add the github-actions ecosystem to Dependabot**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #134: [Maintenance] performance: Hold benchmark projects to TreatWarningsAsErrors (curated benchmarks/.editorconfig)
## Scope

Test projects are now held to `TreatWarningsAsErrors` in Release — the `tests/Directory.Build.props` exemption was removed, and `tests/.editorconfig` carves out the genuinely test-inappropriate analyzer rules per-rule.

Benchmark projects are still exempt — `benchmarks/Directory.Build.props` sets `TreatWarningsAsErrors=false` — because BenchmarkDotNet code trips analyzers heavily (deliberately naive measured code, `GC.GetTotalMemory`, `[Benchmark]` methods invoked by reflection, etc.) and there is no curated `benchmarks/.editorconfig` yet.

Bring benchmarks up to the same bar:
1. Build this repo's benchmark project in Release; collect the analyzer warnings BenchmarkDotNet legitimately triggers.
2. Contribute those genuinely-benchmark-inappropriate rules to a curated `benchmarks/.editorconfig` (canonical in `repo-template`, parallel to `tests/.editorconfig`).
3. Drop `benchmarks/Directory.Build.props` so the benchmark project inherits the root `Directory.Build.props` (`TreatWarningsAsErrors` in Release).
4. Confirm the benchmark project builds clean in Release.

Do this alongside the other benchmark initiatives (P1 — BenchmarkDotNet baseline, P2 — benchmark result graphs).

## Acceptance criteria

- A curated `benchmarks/.editorconfig` exists (canonical), silencing only genuinely-benchmark-inappropriate analyzer rules.
- `benchmarks/Directory.Build.props` removed from this repo.
- This repo's benchmark project builds clean in Release with `TreatWarningsAsErrors`.

## Notes

- maintenance-task — pair with the P1 / P2 benchmark work for this repo.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### LINKED ISSUE #136: [Maintenance] docs: README accuracy fixes (D2)
## Problem

README H1 is `# Wolfgang.D20-Dice` (**dash**), but the actual `<PackageId>` is `Wolfgang.D20.Dice` (**dots**). The repo name uses a dash (`D20-Dice`) but the package follows the canonical dot convention.

## Fix

- Update README H1 to `# Wolfgang.D20.Dice`.
- Search README for `Wolfgang.D20-Dice` and replace with the dot-form where it refers to the package.
- Leave references to the *repo name* (`D20-Dice`) unchanged — those correctly use the dash.

## Source

D2 docs accuracy audit, 2026-05-23.

### LINKED ISSUE #139: [Maintenance] testing: Audit test quality
## Scope

Audit the existing test suite for quality issues that pass CI but reduce real signal: tests that do not assert, weak assertions where stronger ones exist (Assert.True(x != null) -> Assert.NotNull), Assert.Equal(null, ...) instead of Assert.Null, test naming pattern violations (MethodUnderTest_when_condition_expected_result), shared mutable state across test runs, tests of implementation details rather than behavior, hardcoded magic numbers (e.g. 9999999L -> TimeSpan.TicksPerSecond - 1), [Theory] inline data that should be MemberData, missing boundary-case tests for paths the source explicitly handles (MinValue/MaxValue clamps, null-input short-circuits). Run as Phase 4.5 of the code-review skill.

## Acceptance criteria

Each finding categorized as must-fix / should-fix / nit. Must-fix and should-fix items either filed as separate sub-issues or addressed inline. nit items reviewed and either addressed, deferred, or closed as intentional style preference.

## Notes

- Tier-1 foundation task — part of fleet initiative **T4: Audit test quality**.
- Repo-specific scope; assess case by case.
- Board: https://github.com/users/Chris-Wolfgang/projects/6

### PR DESCRIPTION
**Pull request by Chris-Wolfgang**
## Summary

Cuts v0.5.1. Supersedes the stacked canonical PRs (#132 canonical-unprotected, #133 canonical-protected, #140 restore-assemblyversion, #141 D8 fleet sync) — all folded into this one vNext → main merge.

Library public API and runtime behavior are unchanged from v0.5.0; this is canonical CI/docs/metadata work plus the C4 binding-stability fix.

### Added
- **D8** verify-docs-build job in `release.yaml` (deduplicated — the merge produced two identical job keys, fixed in this PR)
- **A1** PublicApiAnalyzers scaffolding (baseline file deferred — see Notes below)
- **CI3** canonical NuGet metadata (Authors/Copyright/SourceLink/snupkg)
- **T3** Stryker mutation-testing workflow
- **T1** coverage report published to docs
- **S1** CodeQL security-extended query pack
- **D6** versions.json preservation guard
- **D7** docs build cache hygiene
- **P2** BenchmarkDotNet → gh-pages chart workflow
- `docs/DOCFX-VERSION-PICKER.md` (the D8 bulk fanout dropped this file on D20-Dice; added directly)

### Changed
- **C1** fleet template-drift sync
- `<Nullable>enable</Nullable>` consolidated into `Directory.Build.props`
- **CI2** Dependabot `github-actions` ecosystem
- **D3** script hardening
- Analyzer `PackageReference`s centralized in `Directory.Build.props`

### Fixed
- **C4** restored explicit `<AssemblyVersion>1.0.0.0</AssemblyVersion>` (4-part form) + prerelease-safe `<FileVersion>` — binding stability across the 0.x line
- Duplicate `verify-docs-build:` job key in `release.yaml` (introduced when origin/main's PR #141 was merged alongside canonical-protected's existing version)
- `.gitattributes` merge conflict resolved by keeping the canonical documented variant

## Notes

- **PublicAPI baseline deferred.** The autogen tool produced an `RS0017`-rejecting line for `TryParse(string?) -> Result<Dice?>` — the analyzer's exact form for nullable generics across assemblies differs from the static-ToDisplayString approximation. Per the per-repo-release-pilot plan, A1 baselines are added later via the IDE "Add to public API" code fix. Scaffolding stays in `Directory.Build.props`; the analyzer self-disables when no `PublicAPI.*.txt` files are present.

## Post-merge

1. Tag `v0.5.1` and publish GitHub Release → `release.yaml` fires
2. Verify NuGet indexing
3. Delete vNext + close superseded PRs (#132, #133, #140, #141)

## Closes (full list across the vNext stack)

Canonical work delivered in this vNext rollup:
- Closes #104 (C1 template drift)
- Closes #107 (C4 version metadata)
- Closes #108 (Nullable enable)
- Closes #112 (P1 BDN baseline)
- Closes #113 (P2 benchmark gh-pages)
- Closes #114 (T1 coverage on docs)
- Closes #116 (T3 Stryker)
- Closes #117 (D — Standardize README)
- Closes #118 (D — CHANGELOG exists)
- Closes #122 (D8 verify-docs-build at release)
- Closes #123 (D6 versions.json preservation)
- Closes #124 (D — version selector canonical assets)
- Closes #125 (S1 CodeQL security-extended)
- Closes #126 (S2 GitHub Actions hardening)
- Closes #128 (A1 PublicApiAnalyzers scaffolding)
- Closes #129 (CI3 NuGet package quality)
- Closes #130 (CI2 Dependabot github-actions)

Stacked PRs merged into vNext (don't auto-close on vNext merges; closing here when vNext lands on main):
- Closes #120 (D2 README accuracy — via PR #145)
- Closes #136 (D2 README accuracy duplicate — via PR #145)
- Closes #121 (D5 audit other markdown — via PR #146)
- Closes #134 (P3 benchmarks .editorconfig — via PR #147)
- Closes #110 (C5 AI code-review pass — via PR #149)
- Closes #139 (T4 test quality — via PR #150)

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -506,17 +562,115 @@ jobs:
...
+            echo "No workload-bearing TFMs in any csproj — skipping dotnet workload restore"
+          fi
+
+      - name: Restore dependencies
+        if: steps.check.outputs.found == 'true'
+        run: dotnet restore
+
+      - name: Build solution (Release)
+        if: steps.check.outputs.found == 'true'
+        run: dotnet build --no-restore --configuration Release
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/release.yaml:631]: The repo does have a root solution — `D20.Dice.sln` at the repo root — so `dotnet restore` / `dotnet build` from the working directory finds it and won't trip MSB1003. Verified locally.

[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/release.yaml:631]: The repo does have a root solution — `D20.Dice.sln` at the repo root — so `dotnet restore` / `dotnet build` from the working directory finds it and won't trip MSB1003. Verified locally.

File: .github/workflows/benchmarks.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v6
+        with:
+          persist-credentials: false
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v5
+        with:
+          dotnet-version: |
+            8.0.x
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/benchmarks.yaml:44]: Good catch — fixed in 8297a92. Added 9.0.x and 10.0.x to the benchmarks.yaml SDK setup so the transitive lib build can target net10.0.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- `docs/DOCFX-VERSION-PICKER.md` documenting the version picker.
+
+### Changed
+
+- **C1** — fleet-wide template-drift sync: workflow files (`pr.yaml`,
+  `release.yaml`, `docfx.yaml`, `codeql.yaml`,
+  `build-all-versions.yaml`, `stryker.yaml`), `.editorconfig`,
+  `BannedSymbols.txt`, `Directory.Build.props`,
+  `tests/Directory.Build.props`, and `.gitattributes` consolidated to
+  the canonical baseline.
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply CHANGELOG.md]: Fixed in e558486 — the CHANGELOG entry now references `tests/.editorconfig` (the actual file the canonical T-tests-editorconfig work installs) rather than `tests/Directory.Build.props` (which doesn't exist in this repo).

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -506,17 +562,115 @@ jobs:
...
+          # netcoreapp3.1 → net10.0). Without these, the docs verify step
+          # fails on repos with broad multi-targeting.
+          dotnet-version: |
+            3.1.x
+            5.0.x
+            6.0.x
+            7.0.x
+            8.0.x
+            9.0.x
+            10.0.x
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/release.yaml:607]: Duplicate of the pr.yaml line 350 comment — same canonical-workflow reasoning. The 'targets netcoreapp3.1' comment in the workflow is generic canonical wording rather than D20-Dice-specific. Fleet-wide cleanup tracked separately. Resolving.

File: .github/workflows/pr.yaml [hunk 5]
```diff
@@ -294,6 +349,20 @@ jobs:
             9.0.x
             10.0.x
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/pr.yaml:350]: Acknowledged — D20-Dice doesn't target netcoreapp3.1 (src csproj is net462/netstandard2.0/net8.0/net10.0; test csproj is net462–net10.0 minus 3.1). The 3.1.x install in this canonical pr.yaml is wasted on this repo. However, the pr.yaml here is a verbatim copy of the canonical `repo-template` workflow which still defensively installs 3.1.x because some downstream repos retain netcoreapp3.1 targets. Removing it locally would create per-repo drift from canonical — flagged for a fleet-wide canonical cleanup (drop 3.1.x from repo-template's pr.yaml + release.yaml + benchmarks.yaml + stryker.yaml SDK lists). Resolving on this PR; the fleet fix lands separately.

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -300,6 +355,7 @@ jobs:
         uses: actions/setup-dotnet@v5
         with:
           dotnet-version: |
+            3.1.x
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/release.yaml:358]: Duplicate of the pr.yaml line 350 comment — same canonical-workflow reasoning. Fleet-wide cleanup tracked separately. Resolving.

File: .github/workflows/pr.yaml [hunk 7]
```diff
@@ -637,6 +718,20 @@ jobs:
...
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/pr.yaml:719]: Duplicate of the comment on line 350 — same reasoning. Wasted on this repo, but canonical pr.yaml content; will be fixed fleet-wide on repo-template, not per-repo. Resolving.

File: .github/workflows/release.yaml [hunk 8]
```diff
@@ -506,17 +562,115 @@ jobs:
...
-    needs: pack-and-validate
+    needs: [pack-and-validate, verify-docs-build]
     if: needs.pack-and-validate.outputs.has-packages == 'true'
     runs-on: windows-latest
     steps:
       - name: Setup .NET
         uses: actions/setup-dotnet@v5
         with:
           dotnet-version: |
+            3.1.x
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/release.yaml:673]: Duplicate of the pr.yaml line 350 comment — same canonical-workflow reasoning. Fleet-wide cleanup tracked separately. Resolving.

File: .github/workflows/release.yaml [hunk 9]
```diff
@@ -23,10 +23,65 @@ jobs:
...
+            $allVersions = ($found.Version | Sort-Object -Unique) -join ', '
+            Write-Error "Release tag '$tagName' (version '$tagVersion') does not match any src csproj version. Found: $allVersions. Bump the csproj <Version> or correct the release tag before re-running."
+            exit 1
+          }
+
       - name: Setup .NET
         uses: actions/setup-dotnet@v5
         with:
           dotnet-version: |
+            3.1.x
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply .github/workflows/release.yaml:84]: Duplicate of the pr.yaml line 350 comment — same canonical-workflow reasoning. Fleet-wide cleanup tracked separately. Resolving.

File: docfx_project/index.md [hunk 10]
```diff
@@ -1,31 +1,31 @@
 ---
-layout: landing
+_layout: landing
 ---
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply docfx_project/index.md:3]: DocFX 2.x actually accepts `_layout` — the underscore-prefix is the reserved-metadata convention DocFX uses (same family as `_appTitle`, `_appFooter` in globalMetadata). DateTime-Extensions has been shipping with `_layout: landing` in production for v1.3.0 / v1.3.1 and renders the landing layout correctly:

  https://chris-wolfgang.github.io/DateTime-Extensions/

Resolving — the underscore-prefix form is the canonical pattern, not a bug.

File: src/Wolfgang.D20.Dice/Wolfgang.D20.Dice.csproj [hunk 11]
```diff
@@ -5,15 +5,20 @@
...
     <LangVersion>latest</LangVersion>
-	  <Version>0.5.0</Version>
+	  <Version>0.5.1</Version>
+	  <!-- Pin AssemblyVersion to a fixed binding-stability baseline so consumers
+	       do not need to recompile / add binding redirects on every minor/patch
+	       bump. Bump only on a deliberate breaking API change. FileVersion +
+	       InformationalVersion (the latter auto-derived from <Version>) carry
+	       the actual release version. -->
+	  <AssemblyVersion>1.0.0.0</AssemblyVersion>
+	  <FileVersion>$([System.Text.RegularExpressions.Regex]::Replace("$(Version)", "[-+].*$", "")).0</FileVersion>
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply src/Wolfgang.D20.Dice/Wolfgang.D20.Dice.csproj]: Good catch — fixed by switching the inner string literals from double quotes to single quotes. Verified the regex still strips prerelease/build-metadata correctly: with `Version=0.5.1-beta.1` the build emits `AssemblyFileVersion=0.5.1.0` as expected. Flagged for fleet-wide canonical follow-up since DbContextBuilder (and the other repos that landed via the same canonical-protected work) carry the same double-quote form.

File: tests/Wolfgang.D20.Dice.Tests/Wolfgang.D20.Dice.Tests.Unit.csproj [hunk 12]
```diff
@@ -2,10 +2,10 @@
 
   <PropertyGroup>
     <TargetFrameworks>net462;net47;net471;net472;net48;net481;net5.0;net6.0;net7.0;net8.0;net9.0;net10.0</TargetFrameworks>
+    <Nullable>enable</Nullable>
 	<Version>0.5.0</Version>
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply tests/Wolfgang.D20.Dice.Tests/Wolfgang.D20.Dice.Tests.Unit.csproj:6]: Acknowledged but can't remove yet. The DBP-hoisted `<Nullable Condition="...$(UsingMicrosoftNETSdk)...">enable</Nullable>` doesn't actually activate under Stage 1 Linux's invocation pattern (`dotnet build --no-restore --framework <fw>`) — same root cause as the explicit `<Nullable>` adds in 8297a92 (src csproj) and 15838ee (benchmarks csproj). When I tried to rely on DBP alone, Stage 1 failed with CS8632 on the test source's 6 nullable annotations (`Dice? other`, `object? obj`, `string? notation`).

This is a fleet-wide canonical concern (every Wolfgang.* repo with nullable annotations will hit it) — flagged for a canonical follow-up on repo-template to either fix the DBP condition or accept that per-csproj overrides are required. Resolving here.

File: src/Wolfgang.D20.Dice/Dice.cs [hunk 13]
```diff
@@ -216,55 +219,39 @@ public bool Equals(Dice? x, Dice? y)
...
     /// </summary>
-    /// <param name="obj">The <see cref="Dice"/> instance for which to generate the hash code.</param>
-    /// <exception cref="ArgumentNullException">Thrown when the <paramref name="obj"/> is null.</exception>
-    /// <returns>int</returns>
+    /// <param name="obj">The <see cref="Dice"/> instance to hash.</param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="obj"/> is null.</exception>
+    /// <returns>A combined hash code of the three identity properties.</returns>
     public int GetHashCode(Dice obj)
     {
         if (obj == null)
```
[the remark this answers is not in the document]
Comment (Chris-Wolfgang) [reply src/Wolfgang.D20.Dice/Dice.cs:241]: Good catch — fixed via PR #149 (the C5 review pass) which is also stacked on vNext. Just pushed an additional commit there switching `obj == null` → `obj is null` in `GetHashCode(Dice obj)` for consistency with the rest of the file. When #149 lands the change propagates to main alongside the vNext merge.
